langlib/langlib-test/src/test/resources/test-src/decimallib_test.bal

```diff
@@ -319,6 +319,123 @@ function testFromStringFunctionWithInvalidValues() {
+    assertEquality(decimal:quantize(123.123, 10E-5), 123.12300d);
+    assertEquality(decimal:quantize(123.123, 1E-1), 123.1d);
+    assertEquality(decimal:quantize(123.123, 0.1E-1), 123.12d);
+    assertEquality(decimal:quantize(123.123, 0.01E-1), 123.123d);
+    assertEquality(decimal:quantize(123.123, 0.001E-1), 123.1230d);
+    assertEquality(decimal:quantize(123.123, 0.0001E-1), 123.12300d);
+    assertEquality(decimal:quantize(263E5, 10E2), 26300000.00d);
+    assertEquality(decimal:quantize(263E5, 10E3), 26300000.000d);
+    assertEquality(decimal:quantize(263E5, 10E4), 26300000.0000d);
+    assertEquality(decimal:quantize(263, 10E-1), 263.0d);
+    assertEquality(decimal:quantize(263, 10E-2), 263.00d);
+    assertEquality(decimal:quantize(263, 10E-3), 263.000d);
+    assertEquality(decimal:quantize(263.2E2, 10E-1), 26320.0d);
+    assertEquality(decimal:quantize(263.2E2, 10E-2), 26320.00d);
+    assertEquality(decimal:quantize(2.632E2, 10E-1), 263.2d);
+    assertEquality(decimal:quantize(2.632E2, 10E-2), 263.20d);
+    assertEquality(decimal:quantize(2.632E2, 10E-3), 263.200d);
+    assertEquality(decimal:quantize(2.632E2, 10E-31), 263.2000000000000000000000000000000d);
+    assertEquality(decimal:quantize(-0.1, 1), 0d);
+    assertEquality(decimal:quantize(2632E-2, 10E-4), 26.3200d);
+    assertEquality(decimal:quantize(2632E-2, 10E-3), 26.320d);
+    assertEquality(decimal:quantize(2632E-2, 10E-2), 26.32d);
+    assertEquality(decimal:quantize(2632E-2, 10E-1), 26.3d);
+    assertEquality(decimal:quantize(2632E-2, 10E0), 26d);
+    assertEquality(decimal:quantize(2632E-2, 10E1), 3E1d);
+    assertEquality(decimal:quantize(2632E-2, 10E2), 0d);
+    assertEquality(decimal:quantize(2632E-2, 10E3), 0d);
+    assertEquality(decimal:quantize(0.26E-2, 10E-6), 0.002600d);
+    assertEquality(decimal:quantize(0.26E-2, 10E-5), 0.00260d);
+    assertEquality(decimal:quantize(0.26E-2, 10E-4), 0.0026d);
+    assertEquality(decimal:quantize(0.26E-2, 10E-3), 0.003d);
+    assertEquality(decimal:quantize(0.26E-2, 10E-2), 0.00d);
+    assertEquality(decimal:quantize(0.26E-2, 10E-1), 0d);
+    assertEquality(decimal:quantize(0.26E-2, 10E0), 0d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E-6), -0.002600d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E-5), -0.00260d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E-4), -0.0026d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E-3), -0.003d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E-2), -0.00d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E-1), 0d);
+    assertEquality(decimal:quantize(-0.26E-2, 10E0), 0d);
+}
+
+function testQuantizeFunctionWithInvalidOutput() {
+    decimal|error a1 = trap decimal:quantize(12.3, 2E-35);
+    assertEquality(true, a1 is error);
+    if a1 is error {
+        assertEquality("{ballerina/lang.decimal}InvalidOperation", a1.message());
+    }
+
+    a1 = trap decimal:quantize(1, 2E-35);
```

Comment:
@chiranSachintha, can you explain why this is an invalid operation.

Reply:
```
public function main() {
    decimal k = 1.23454994949444444444444444444444444484848844444444444444444444499944;

    io:println(k);
}
```
If we consider the above example the output is `1.234549949494444444444444444444444`. The reason for that is the maximum number of digits in the significand is 34 for the 128-bit decimal (radix 10) floating-point number. In the above example, the significand is 35.

Reply:
Understood, thank you for the clarification.